examples/schoolyard.jl

```diff
@@ -0,0 +1,138 @@
+# The bond between pairs is chosen at random between 0 and 1, with a bond of 1 being the strongest.
+# If the bond is *friendly*, agents wish above all else to be near their *friend*.
+# Bonds that are *unfriendly* see students moving as far away as possible from their *foe*.
+
+# ## Initialising the model
+
+function schoolyard(;
+    numStudents = 50,
+    teacher_attractor = 0.2,
+    noise = 0.1,
+    max_force = 1.5,
+)
+    model = ABM(
+        Student,
+        ContinuousSpace(2; extend = (100, 100));
+        properties = Dict(
+            :teacher_attractor => teacher_attractor,
+            :noise => noise,
+            :buddies => SimpleWeightedGraph(numStudents),
+            :max_force => max_force,
+        ),
+    )
+    for student in 1:numStudents
+        ## Students begin near the school building
+        add_agent!(model.space.extend .* 0.5 .+ Tuple(rand(2)) .- 0.5, model)
+
+        ## Add one friend and one foe to the social network
+        friend = rand(filter(s -> s != student, 1:numStudents))
+        add_edge!(model.buddies, student, friend, rand())
+        foe = rand(filter(s -> s != student, 1:numStudents))
+        add_edge!(model.buddies, student, foe, -rand())
+    end
+    model
+end
+
+# Our model contains the `buddies` property, which is our LightGraphs.jl undirected, weighted graph.
+# As we can see in the loop, we choose one `friend` and one `foe` at random for each `student` and
+# assign their relationship as a weighted edge on the graph.
+
+# ## Movement dynamics
+
+distance(pos) = sqrt(pos[1]^2 + pos[2]^2)
+scale(L, force) = (L / norm(force)) .* force
+
+function agent_step!(student, model)
+    ## place a teacher in the center of the yard, so we don’t go too far away
+    teacher = (model.space.extend .* 0.5 .- student.pos) .* model.teacher_attractor
+
+    ## add a bit of randomness
+    noise = model.noise .* (Tuple(rand(2)) .- 0.5)
+
+    ## Adhere to the social network
+    network = model.buddies.weights[student.id, :]
+    tidxs, tweights = findnz(network)
+    network_force = (0.0, 0.0)
+    for (widx, tidx) in enumerate(tidxs)
+        buddiness = tweights[widx]
+        force = (student.pos .- model[tidx].pos) .* buddiness
+        if buddiness >= 0
+            ## The further I am from them, the more I want to go to them
+            if distance(force) > model.max_force # I'm far enough away
+                force = scale(model.max_force, force)
+            end
+        else
+            ## The further I am away from them, the better
+            if distance(force) > model.max_force # I'm far enough away
+                force = (0.0, 0.0)
+            else
+                L = model.max_force - distance(force)
+                force = scale(L, force)
+            end
+        end
+        network_force = network_force .+ force
+    end
+
+    ## Add all forces together to assign the students next position
+    student.pos = student.pos .+ noise .+ teacher .+ network_force
```

Comment:
here you should use `move_agent!(student, newpos, model)` otherwise the space structure won't be able to keep track of the change in position.

Reply:
That's what the update_space! call does, but sure&mdash;I can change to move_agent! here.

Reply:
No, I can't do this. Need to use update_space! here, since move_agent!(student, newpos, model) doesn't exist for continuous space. That's why the example fails.